modules/afsocket/afsocket-dest.c

```diff
@@ -696,10 +696,13 @@ afsocket_dd_notify(LogPipe *s, gint notify_code, gpointer user_data)
     case NC_WRITE_ERROR:
       log_writer_reopen(self->writer, NULL);
 
+      char *server_address = self->dest_addr ? g_sockaddr_format(self->dest_addr, buf, sizeof(buf), GSA_FULL) : "";
       msg_notice((notify_code == NC_CLOSE) ? "Syslog connection closed" : "Syslog connection broken",
                  evt_tag_int("fd", self->fd),
-                 evt_tag_str("server", g_sockaddr_format(self->dest_addr, buf, sizeof(buf), GSA_FULL)),
+                 evt_tag_str("server", server_address),
```

Comment:
Do I get it right, that this is a kept-alive connection that we could resolve when the connection was initially established, but then when the reload happens we cannot resolve it anymore? Is that the scenario we are addressing here?

Maybe we could report the unresolvable name in this case, instead of an empty string.

But if this is the case we might need to be fully operational while self->dest_addr is NULL. How come our assertion does not trigger?

Reply:
btw: we seem to have a wrapper struct that we store in PersistConfig, so we could as well save dest_addr there.

it'd be strange that we don't know our own dest address. Or we could as well do a getpeername() call and restore the destination address that way.

Reply:
I've saved the `dest_addr` across reloads. It was tricky to reuse it (AFInetDestDriver now has to check for an open connection in setup_address to avoid blindly dropping it).

I haven't replaced the LogWriter with LogProto in the "reload_item" (i.e. the object in CfgPersist, which we would like to reuse after reload) yet.

I would like to check if it's possible, as it **could remove the need for the new `log_writer_steal_proto()` API**.
(p.s. I still think the asymmetrical `log_writer_init` is not okay, but that's independent of that).